airbyte-integrations/connectors/source-bigquery/source_bigquery/source.py

```diff
@@ -82,31 +113,105 @@ def check_connection(self, logger, config: Mapping[str, Any]) -> Tuple[bool, any
+        normal_streams = []
+        concurrent_streams = []
+
+        for stream in streams:
+                dataset_id, table_id = stream['parent_stream'].split(".")
+                where_clause = stream["where_clause"]
+                stream_name = stream["name"]
+                if sync_method == "Standard":
+                    table_obj = IncrementalQueryResult(project_id, dataset_id, table_id, stream_name, where_clause, fallback_start=fallback_start, slice_range=slice_range, authenticator=self._auth)
+                else:
+                    try:
+                        table_obj = TableChangeHistory(project_id, dataset_id, table_id, stream_name, where_clause=where_clause, fallback_start=change_history_start, slice_range=slice_range, authenticator=self._auth)
+                        next(table_obj.read_records(sync_mode=SyncMode.full_refresh))
+                    except Exception as e:
+                        self.logger.warn(str(e))
+                        table_obj = IncrementalQueryResult(project_id, dataset_id, table_id, stream_name, where_clause, fallback_start=fallback_start, slice_range=slice_range, authenticator=self._auth)
+                if isinstance(table_obj, TableChangeHistory):
+                    concurrent_streams.append(table_obj.stream)
+                else:
+                    normal_streams.append(table_obj.stream)
+        for dataset in BigqueryDatasets(project_id=project_id, authenticator=self._auth).read_records(sync_mode=SyncMode.full_refresh):
             dataset_id = dataset.get("datasetReference")["datasetId"]
             # list and process each table under each base to generate the JSON Schema
-            for table_info in BigqueryTables(dataset_id=dataset_id, project_id=config["project_id"], authenticator=self._auth).read_records(sync_mode=SyncMode.full_refresh):
+            for table_info in BigqueryTables(dataset_id=dataset_id, project_id=project_id, authenticator=self._auth).read_records(sync_mode=SyncMode.full_refresh):
                 table_id = table_info.get("tableReference")["tableId"]
                 if sync_method == "Standard":
-                    table_obj = IncrementalQueryResult(config["project_id"], dataset_id, table_id, authenticator=self._auth)
+                    table_obj = IncrementalQueryResult(project_id, dataset_id, table_id, fallback_start=fallback_start, slice_range=slice_range, authenticator=self._auth)
                 else:
                     try:
-                        table_obj = TableChangeHistory(config["project_id"], dataset_id, table_id, authenticator=self._auth)
+                        table_obj = TableChangeHistory(project_id, dataset_id, table_id, fallback_start=change_history_start, slice_range=slice_range, authenticator=self._auth)
                         next(table_obj.read_records(sync_mode=SyncMode.full_refresh))
                     except Exception as e:
```

Comment:
- Updated here https://github.com/artefactory/airbyte/pull/22/commits/985421a881c5e333d7106455bb8d7beff63d3fb1
- In one of the demos we agreed that we should just get the stream without CDC in case there's no change history. Your suggestion makes sense too because with this behaviour the user might think that all their streams are using change history, this can be solved by adding it in the documentation so that users can know which streams are in CDC mode